### DISCUSSION TITLE
collect usage with yabeda-statsd

### DISCUSSION THREAD

**Discussion by mbajur** [yabeda-rb/yabeda#17] [Q&A] [answered]
Hello!
I was trying to use a `collect` block to gather memory usage of my app processes but can't really make it work. My config (rails initializer) look as follows:

```
Yabeda.configure do
  group :app do
    gauge :process_rss
  end

  collect do
    app.process_rss.set(
      { process_name: $PROGRAM_NAME },
      `ps -o rss= -p #{Process.pid}`.to_i
    )
  end
end

Yabeda::Statsd.start
```

But no gauge is being sent to statsd deamon. However, running
```
Yabeda.app.process_rss.set({ process_name: $PROGRAM_NAME }, `ps -o rss= -p #{Process.pid}`.to_i)
```
manually works as expected and sends the gauge metric so it seems to be something wrong about the `collect` block itself. Should it be working out of the box or do i need to take any extra steps to enable it?

My stack
- Rails 6.0.3.2
- Ruby 2.7.0
- Yabeda 0.8.0
- yabeda-rails 0.7.1
- yabeda-statsd 0.1.2

Thanks in advance for any clues!

**Comment by Envek** [comment, marked as answer]
I can see that in [yabeda-statsd] collection of metrics defined in `collect` blocks is done in separate `start_exporter` method here:

 https://github.com/asusikov/yabeda-statsd/blob/c248f138f7229a50f570ca5a07394fb079781138/lib/yabeda/statsd.rb#L31-L39

Unfortunately this method doesn't seem to be called from anywhere, so I suppose that you have to call it explicitly right after `start`:

```ruby
Yabeda::Statsd.start
Yabeda::Statsd.start_exporter
```

Please open issue in the [yabeda-statsd] repo as it is a third-party adapter I don't have control over.

[yabeda-statsd]: https://github.com/asusikov/yabeda-statsd "Yabeda Statsd adapter. Collect and send custom metrics from Ruby apps to your Statsd server "

**Reply by mbajur** [reply]
That's exactly a piece of information i was missing. Thank you! It works now :)
